### ISSUE TITLE
[SPFx-Heft-Plugins][SPFx 1.13.1] elementManifests path resolving differently on Windows and Linux when referencing external file path

### ISSUE THREAD

**Issue by NickSevens** [SharePoint/sp-dev-docs#7689]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [X] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Describe the bug / error

We are working on a solution consisting of multiple SPFx projects, exposed as separate NPM packages. For our customers, we gather these NPM packages into a single SPFx package, joined together in a `yarn workspaces` project.

This results in a project structure like this:

```
<root>
├───apps
│   └───demoapp
│       ├───config
│       │   ├───package-solution.json
│       │   └───...
│       ├───gulpfile.js
│       └───...
├───node_modules
│   └───@mycompany
│       └───appcustomizer
│           └───...
├───packages
│   └───appcustomizer
│       ├───config
│       ├───sharepoint
│       │   └───assets
│       │       └───elements.xml
│       └───src
└───package.json
```

In the example above, in the `demoapp` SPFx package, one of the packages we are referencing contains an Application Customizer.

For this Application Customizer, we reference the `elements.xml` from the _source_ NPM package, located in its `node_modules` directory. (`<root>/node_modules/@mycompany/appcustomizer/sharepoint/assets/elements.xml`)

```
...
"features": [
  {
    "title": "Feature title",
    "description": "...",
    "id": "...",
    "version": "1.1.0.0",
    "assets": {
      "elementManifests": ["../../../../../node_modules/@my-company/appcustomizer/sharepoint/assets/elements.xml"]
    }
  }
],
... 
```

The issue is however, than when calling `gulp package-solution`, depending on the environment this path should be **different** to work:
* on **Windows** this path should be `../../../../../node_modules/...` (5x `../`)
* on **Linux** this path should be `../../../../node_modules/...` (4x `../`)

### Debugging results

The issue seems to originate in the `@microsoft/spfx-heft-plugins` package, more specifically in `lib/plugins/packageSolutionPlugin/packageSolution/utils/normalizeSolutionDefinition.js`.

In this file, the function `appendBasePathToFeatureFiles` uses `path.posix.join(basePath, filename)`.

It seems that, this function doesn't work correctly when referencing such relative paths.
I checked the `basePath` and `fileName` parameters, which return the following:

On **Windows**:
```
basePath: C:\...\apps\connect-demo/sharepoint/assets
filename: ../../../../../node_modules/@dw-connect/appheader/sharepoint/assets/elements.xml
```

On **Linux**:
```
basePath: /mnt/c/.../apps/connect-demo/sharepoint/assets
filename: ../../../../../node_modules/@dw-connect/appheader/sharepoint/assets/elements.xml
```

When running `path.posix.join` on those parameters, depending on the environment, these yield a different result.
Running `path.join`, however, would return the correct result on both Windows and Linux, for the same parameters.

**Comment by AJIXuMuK** [comment]
Hi @NickSevens - sorry, I'm not so good with Linux. But why does the relative path differ? `posix` namespace is initially created to work with Linux systems.
Is the difference in how `yarn workspaces` generate the structure?

**Comment by NickSevens** [comment]
Hi @AJIXuMuK, the path itself is not the issue. The generated structure is exactly the same on linux and windows.
However, it's the `path.posix.join` function which returns a different relative structure on either system.
Changing it to `path.join` (so not the `posix` version) does work though.

I'm not good with Linux either, but we run our builds on an azure devops linux agent. That's where the issue surfaced. I can also reproduce the same on the Ubuntu WSL prompt.

I'm not sure why the `posix` part is needed at your end in the plugin? Is it an option to simply change this in a future release? :)

We'd love to see a fix so we can upgrade our projects to the newest SPFx versions - as I assume the future versions will use this build system as well.

PS: I'll try to create a reproduction repo for you guys, if it helps?

**Comment by AJIXuMuK** [comment]
@NickSevens - we will make the changes on how the resolution works. You will be able to use the standard node_modules resolution for your scenario.
But it will be included in 1.15 (early beta) only.
For now I'd just play around to find out if some workarounds help (maybe adding `./` at the beginning of the elements file path)

**Comment by patmill** [comment]
This has been fixed and released in the latest 1.15 beta
